### PR TITLE
[SELS-TASK][BE] Category Management

### PR DESCRIPTION
**Pull request by louisepujante**
### Asana
-[link](https://app.asana.com/0/1201426075515354/1201464141731410/f)

### Commands
-cd backend > composer install > php artisan migrate:fresh --seed

### Pre-conditions
-copy .env.example and rename to .env then provide local db credentials

viewing all / viewing specific category:
-must be logged in

create, update, delete:
-must be an admin
-name is required and must be unique, description field must be present but can be empty

### Expected Output
viewing all / viewing specific category:
-can view all or specific categories

create, update, delete:
-admin is able to create, update or delete category if details are correct

### CODE REVIEW

File: backend/app/Providers/AuthServiceProvider.php [hunk 1]
```diff
@@ -25,6 +26,9 @@ public function boot()
     {
         $this->registerPolicies();
 
-        //
+        Gate::define('can_modify', function ($user) {
+            // echo($user);
```
Comment (kimjo77) [inline backend/app/Providers/AuthServiceProvider.php:30]: remove `echo` if not longer needed

File: backend/app/Exceptions/Handler.php [hunk 2]
```diff
@@ -42,7 +42,14 @@ class Handler extends ExceptionHandler
     public function register()
     {
         $this->renderable(function (Exception $e, $request) {
-            if ($request->wantsJson()) {                
+            if ($request->wantsJson()) {           
+                 if($e instanceof NotFoundHttpException) {
+                    // echo(get_class($e));
```
Comment (kimjo77) [inline backend/app/Exceptions/Handler.php:47]: remove `echo` if not longer needed

File: backend/app/Http/Controllers/Category/CategoryController.php [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    }
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(Request $request)
+    {
+        // echo(auth()->user());
```
Comment (kimjo77) [inline backend/app/Http/Controllers/Category/CategoryController.php:34]: remove `echo` if not longer needed

**Review by kimjo77 — APPROVED** [review]

**Comment by louisepujante** [comment]
Merging this branch to main since changes have been approved

**Pull request merged**